provider/aws/aws.go

```diff
@@ -440,12 +478,13 @@ func (p *AWSProvider) records(ctx context.Context, zones map[string]*route53.Hos
 
 	for _, z := range zones {
 		params := &route53.ListResourceRecordSetsInput{
-			HostedZoneId: z.Id,
+			HostedZoneId: z.zone.Id,
 			MaxItems:     aws.String(route53PageSize),
 		}
 
-		if err := p.client.ListResourceRecordSetsPagesWithContext(ctx, params, f); err != nil {
-			return nil, errors.Wrapf(err, "failed to list resource records sets for zone %s", *z.Id)
+		client := p.clients[z.profile]
+		if err := client.ListResourceRecordSetsPagesWithContext(ctx, params, f); err != nil {
+			return nil, errors.Wrapf(err, "failed to list resource records sets for zone %s using aws profile %q", *z.zone.Id, z.profile)
```

Comment:
You should not use errors package to wrap anymore.
`fmt.Errorf("...: %w", ..., err)` is the way

Reply:
agree :+1:

Reply:
agree :+1: